# BioHansel Manuscript Supplementary Data Repository.

This repository contains supplementary data for the BioHansel manuscript organized by section in which the supplementary data is referenced.

The directory structure of this repo follows the structure of the BioHansel manuscript.

## Methods

### Canonical SNP genotyping schemas

BioHansel scheme development documentation: https://bio-hansel.readthedocs.io/en/readthedocs/user-docs/genotyping_schemes.html#creating-a-genotyping-scheme

- [Appendix I Additional info BioHansel schema development and adaptation.docx](./Methods/Canonical%20SNP%20genotyping%20schemas/Appendix_I_Additional_info_BioHansel_schema_development_and_adaptation.docx)
- [Table S1 Backbone positions 5 schemes.xlsx](./Methods/Canonical%20SNP%20genotyping%20schemas/Table_S1_Backbone_positions_5_schemas.xlsx)

#### BioHansel Schemas

- [Schema 1 SH_scheme_v0.5.0.fasta](./Methods/Canonical%20SNP%20genotyping%20schemas/BioHansel%20Schemas/Schema%201%20SH_scheme_v0.5.0.fasta)
- [Schema 2 SE_scheme_v1.0.7.fasta](./Methods/Canonical%20SNP%20genotyping%20schemas/BioHansel%20Schemas/Schema%202%20SE_scheme_v1.0.7.fasta)
- [Schema 3 ST_scheme_v0.5.5.fasta](./Methods/Canonical%20SNP%20genotyping%20schemas/BioHansel%20Schemas/Schema%203%20ST_scheme_v0.5.5.fasta)
- [Schema 4 Typhi_scheme_v1.2.0.fasta](./Methods/Canonical%20SNP%20genotyping%20schemas/BioHansel%20Schemas/Schema%204%20Typhi_scheme_v1.2.0.fasta)
- [Schema 5 tb_speciation_scheme_v1.0.5.fasta](./Methods/Canonical%20SNP%20genotyping%20schemas/BioHansel%20Schemas/Schema%205%20tb_speciation_scheme_v1.0.5.fasta)

#### Population structure backbones

- [Fig S1a Enteritidis backbone v1.0.7.pdf](./Methods/Canonical%20SNP%20genotyping%20schemas/Population_structure_backbones/Fig_S1a_Enteritidis_backbone_v1.0.7.pdf)
- [Fig S1b Heidelberg backbone v0.5.0.pdf](./Methods/Canonical%20SNP%20genotyping%20schemas/Population_structure_backbones/Fig_S1b_Heidelberg_backbone_v0.5.0.pdf)
- [Fig S1c Typhimurium backbone v0.5.5.pdf](./Methods/Canonical%20SNP%20genotyping%20schemas/Population_structure_backbones/Fig_S1c_Typhimurium_backbone_v0.5.5.pdf)
- [Fig S1d Typhi backbone v1.2.0.pdf](./Methods/Canonical%20SNP%20genotyping%20schemas/Population_structure_backbones/Fig_S1d_Typhi_backbone_v1.2.0.pdf)
- [Fig S1e MTB backbone v1.0.5.pdf](./Methods/Canonical%20SNP%20genotyping%20schemas/Population_structure_backbones/Fig_S1e_MTB_backbone_v1.0.5.pdf)
- [Caption trees Fig S1a-e.txt](./Methods/Canonical%20SNP%20genotyping%20schemas/Population_structure_backbones/Caption_trees_Fig_S1a-e.txt)

### BioHansel

#### Examples of BioHansel output files

- [Table S2a BioHansel Simple Summary genome NC_011083 tech results.tab](./Methods/BioHansel/BioHansel%20output%20files/Table%20S2a%20BioHansel_Simple_Summary_genome_NC_011083_tech_results.tab)
- [Table S2b BioHansel Results Summary genome NC_011083 results.tab](./Methods/BioHansel/BioHansel%20output%20files/Table%20S2b%20BioHansel_Results_Summary_genome_NC_011083_results.tab)
- [Table S2c BioHansel Detailed K-mer Results genome NC_011083 match results.tab](./Methods/BioHansel/BioHansel%20output%20files/Table%20S2c%20BioHansel_Detailed_K-mer_Results_genome_NC_011083_match_results.tab)
- [Table S2d BioHansel Simple Summary unassembled reads SH outbreaks tech results.tab](./Methods/BioHansel/BioHansel%20output%20files/Table%20S2d%20BioHansel_Simple_Summary_unassembled_reads_SH_outbreaks_tech_results.tab)
- [Table S2e BioHansel Results Summary unassembled reads SH outbreaks results.tab](./Methods/BioHansel/BioHansel%20output%20files/Table%20S2e%20BioHansel_Results_Summary_unassembled_reads_SH_outbreaks_results.tab)
- [Table S2f BioHansel Detailed K-mer Results unassembled reads SH outbreaks match results.tab](./Methods/BioHansel/BioHansel%20output%20files/Table%20S2f%20BioHansel_Detailed_K-mer_Results_unassembled_reads_SH_outbreaks_match_results.tab)

#### BioHansel metadata tables

- [Table S3a tb_speciation_scheme_v1.0.5_metadata_Coll_et_al_genotypes.txt](./Methods/BioHansel/BioHansel%20metadata%20tables/Table%20S3a%20tb_speciation_scheme_v1.0.5_metadata_Coll_et_al_genotypes.txt)
- [Table S3b Typhi_scheme_v1.2.0_metadata_Wong_et_al_genotypes.txt](./Methods/BioHansel/BioHansel%20metadata%20tables/Table%20S3b%20Typhi_scheme_v1.2.0_metadata_Wong_et_al_genotypes.txt)

#### Source Code

Source code available at https://github.com/phac-nml/biohansel/

### Analysis of retrospective outbreaks with BioHansel

- [Table S4 SH Outbreaks Bekal and coll.tab](./Methods/Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/Table_S4_SH_Outbreaks_Bekal_and_coll.tab)
- [Reference genome for SNVPhyl analysis of SH outbreaks: SH_ref_SL476_NC_011083.1.fasta](./Methods/Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SH_ref_SL476_NC_011083.1.fasta)

### BioHansel SNP detection validation

#### BioHansel and Snippy SNP detection concordance for *S*. Typhi

See https://github.com/peterk87/nf-biohansel-snippy-comparison/blob/master/main.nf

- [Table S5 List of 1000 Salmonella Typhi Accessions.txt](./Methods/BioHansel%20SNP%20detection%20validation/SalmonellaTyphi%20WGS%20dataset%20construction/Table_S5_List_of_1000_Salmonella_Typhi_Accessions.txt)
- [Worksheet 1 Detailed parameter settings for all programs used for validation](./Methods/BioHansel%20SNP%20detection%20validation/Supplementary%20Worksheet%201.pdf)
- [Raw Nextflow workflow results for Jupyter Notebook](./Methods/BioHansel%20SNP%20detection%20validation/Concordance%20Assessment%20-%20S.%20Typhi/bh-snippy-compare-1000genomes-S2-accessions.csv/bh-snippy-compare-1000genomes-S2-accessions.csv)
- [Jupyter Notebook to tabulate NextFlow workflow results](./Methods/BioHansel%20SNP%20detection%20validation/Concordance%20Assessment%20-%20S.%20Typhi/py-compile-biohansel-snippy-comparison-results.ipynb)
- [HTML version of Jupyter Notebook to tabulate NextFlow workflow results](./Methods/BioHansel%20SNP%20detection%20validation/Concordance%20Assessment%20-%20S.%20Typhi/py-compile-biohansel-snippy-comparison-results.html)

### Contamination Detection

#### BioHansel and Snippy SNP calling concordance on artificial contamination datasets

- [Table S6 List of genomes used to generate artificial contamination datasets](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20basecalling%20comparison/Table_S6_List_of_reference_genomes.txt)
- [Worksheet 2 Bash and Seqtk Commands.xlsx](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20basecalling%20comparison/Worksheet%202%20Bash%20and%20Seqtk%20Commands.xlsx)

#### Reference genomes sequences used for Snippy analysis

- [AE006468.2_ref_ST_LT2.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/AE006468.2_ref_ST_LT2.fasta)
- [CP012921.1_ref_SH.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/CP012921.1_ref_SH.fasta)
- [NC_000962.3_ref_MTB_H37Rv.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/NC_000962.3_ref_MTB_H37Rv.fasta)
- [NC_003198.1_ref_Typhi_CT18.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/NC_003198.1_ref_Typhi_CT18.fasta)
- [NC_011294.1_ref_SE_P125109.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/NC_011294.1_ref_SE_P125109.fasta)

For Python scripts used to extract the detected variants, see https://github.com/jrober84/bio_hansel_benchmarking


#### Contamination detection at different coverage cut offs

#### Selecting representative datasets for the five genotyping schemas

- [Table S7 List of ST accessions for generation of contaminated datasets.txt](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20contamination%20detection/Table_S7_List_of_ST_accessions_for_generation_of_contaminated_datasets.tab)
- [Table S8 List of SE accessions for generation of contaminated datasets.txt](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20contamination%20detection/Table_S8_List_of_SE_accessions_for_generation_of_contaminated_datasets.tab)
- [Table S9 List of Typhi accessions for generation of contaminated datasets.txt](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20contamination%20detection/Table_S9_List_of_Typhi_accessions_for_generation_of_contaminated_datasets.tab)
- [Table S10 List of MTB accessions for generation of contaminated datasets.txt](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20contamination%20detection/Table_S10_List_of_MTB_accessions_for_generation_of_contaminated_datasets.tab)
- [Table S11 List of SH accessions for generation of contaminated datasets.txt](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20contamination%20detection/Table_S11_List_of_SH_accessions_for_generation_of_contaminated_datasets.tab)

#### Scripts

- [Script 1 create_reads.sh](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20contamination%20detection/Script%201%20create_reads.sh)
- [Script 2 contaminate.sh](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20contamination%20detection/Script%202%20contaminate.sh)
- [Appendix II Additional info Generation of artificially contaminated datasets.pdf](./Methods/Contamination%20Detection/Artificial%20contamination%20datasets%20for%20contamination%20detection/Appendix%20II%20Supplementary%20methods%20for%20Contamination%20Process.pdf)

### BioHansel compute performance

See https://github.com/peterk87/nf-biohansel-sra-benchmark for BioHansel and Snippy benchmarking workflow

#### Selection of datasets and construction of artificial sequencing data

- [Table S12 List of 1017 Salmonella Enteritidis strains NextFlow workflow.txt](./Methods/BioHansel%20compute%20performance/Datasets/Table_S12_List_of_1017_Salmonella_Enteritidis_strains_NextFlow_workflow.txt)

##### Selection of datasets for additional benchmarking

- [Table S13 List of 1000 Salmonella Enteritidis strains.tab](./Methods/BioHansel%20compute%20performance/Datasets/Table_S13_List_of_1000_Salmonella_Enteritidis_strains.tab)
- [Table S14 List of 1000 Salmonella Heidelberg strains.tab](./Methods/BioHansel%20compute%20performance/Datasets/Table_S14_List_of_1000_Salmonella_Heidelberg_strains.tab)
- [Table S15 List of 1000 Salmonella Typhi strains.tab](./Methods/BioHansel%20compute%20performance/Datasets/Table_S15_List_of_1000_Salmonella_Typhi_strains.tab)
- [Table S16 List of 1000 Salmonella Typhimurium strains.tab](./Methods/BioHansel%20compute%20performance/Datasets/Table_S16_List_of_1000_Salmonella_Typhimurium_strains.tab)

#### Generation of artificial sequencing data representing various genome coverage

##### Reference genomes sequences

- [AE006468.2_ref_ST_LT2.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/AE006468.2_ref_ST_LT2.fasta)
- [CP012921.1_ref_SH.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/CP012921.1_ref_SH.fasta)
- [NC_000962.3_ref_MTB_H37Rv.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/NC_000962.3_ref_MTB_H37Rv.fasta)
- [NC_003198.1_ref_Typhi_CT18.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/NC_003198.1_ref_Typhi_CT18.fasta)
- [NC_011294.1_ref_SE_P125109.fasta](./Methods/Contamination%20Detection/Concordance%20Assessment/Reference%20genomes/NC_011294.1_ref_SE_P125109.fasta)

#### Determination of runtime and memory usage

- [Jupyter Notebook to tabulate NextFlow workflow results](./Methods/BioHansel%20compute%20performance/Determination%20of%20runtime%20and%20memory%20usage/py-biohansel-vs-snippy-benchmarking-results.ipynb)
- [HTML version of Jupyter Notebook](./Methods/BioHansel%20compute%20performance/Determination%20of%20runtime%20and%20memory%20usage/py-biohansel-vs-snippy-benchmarking-results.html)
- [NextFlow workflow output file](./Methods/BioHansel%20compute%20performance/Determination%20of%20runtime%20and%20memory%20usage/trace.csv)

## Results and Discussion

### BioHansel quickly and accurately genotypes isolates from WGS data

#### BioHansel results SH outbreaks

- [Table S17a BioHansel tech results SH outbreaks.txt](./Results%20and%20Discussion/BioHansel%20quickly%20and%20accurately%20genotypes%20isolates%20from%20WGS%20data/Outbreak%20Analysis/BioHansel%20results%20for%20SH%20outbreaks%20dataset/Table%20S17a%20BioHansel_raw_datasets_tech_results_SH_outbreaks.tab)
- [Table S17b BioHansel results SH outbreaks.txt](./Results%20and%20Discussion/BioHansel%20quickly%20and%20accurately%20genotypes%20isolates%20from%20WGS%20data/Outbreak%20Analysis/BioHansel%20results%20for%20SH%20outbreaks%20dataset/Table%20S17b%20BioHansel_raw_datasets_results_SH_outbreaks.tab)
- [Table S17c BioHansel match results SH outbreaks.txt](./Results%20and%20Discussion/BioHansel%20quickly%20and%20accurately%20genotypes%20isolates%20from%20WGS%20data/Outbreak%20Analysis/BioHansel%20results%20for%20SH%20outbreaks%20dataset/Table%20S17c%20BioHansel_raw_datasets_match_results_SH_outbreaks.tab)

#### SNVPhyl run SH outbreaks

- [Appendix III SH outbreak analysis by SNVPHYL-results.zip](./Results%20and%20Discussion/BioHansel%20quickly%20and%20accurately%20genotypes%20isolates%20from%20WGS%20data/Outbreak%20Analysis/Appendix%20III%20SH_outbreak_analysis_by_SNVPHYL-results.zip)

#### BioNumerics results for SH outbreaks

- [Figure S2 BioNumerics Dendrogram](./Results%20and%20Discussion/BioHansel%20quickly%20and%20accurately%20genotypes%20isolates%20from%20WGS%20data/Outbreak%20Analysis/SH_outbreak_analysis_with_BioNumerics_wgMLST/Figure%20S2%20BioNumerics%20dendrogram.pdf)
- [Table S18 BioNumerics wgMLST similarity matrix.xlsx](./Results%20and%20Discussion/BioHansel%20quickly%20and%20accurately%20genotypes%20isolates%20from%20WGS%20data/Outbreak%20Analysis/SH_outbreak_analysis_with_BioNumerics_wgMLST/Table%20S18%20BioNumerics%20wgMLST%20similarity%20matrix.xlsx)
- [Table S19 BioNumerics wgMLST character data.xlsx](./Results%20and%20Discussion/BioHansel%20quickly%20and%20accurately%20genotypes%20isolates%20from%20WGS%20data/Outbreak%20Analysis/SH_outbreak_analysis_with_BioNumerics_wgMLST/Table%20S19%20BioNumerics%20wgMLST%20character%20data.xlsx)

#### Outbreak Analysis results

- [Table S4 SH Outbreaks Bekal and coll.tab](./Methods/Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/Table_S4_SH_Outbreaks_Bekal_and_coll.tab)
- [Table S20 Heidelberg backbone tree sequences and genome positions BioHansel scheme v0.5.0.txt](./Results%20and%20Discussion/BioHansel%20quickly%20and%20accurately%20genotypes%20isolates%20from%20WGS%20data/Table_S20_Heidelberg_backbone_BioHansel_scheme%20v0.5.0.tab)
- [Table S21 SH outbreaks SNV distance matrix.txt](./Results%20and%20Discussion/BioHansel%20quickly%20and%20accurately%20genotypes%20isolates%20from%20WGS%20data/Table_S21_SH_outbreaks_SNV_distance_matrix.tab)

### BioHansel’s genotyping results have high concordance with traditional SNP calling workflows

- [Table S22 BioHansel vs Snippy Coverage Comparison 1000 Typhi datasets](./Results%20and%20Discussion/BioHansel's%20genotyping%20results%20have%20high%20concordance%20with%20traditional%20SNP%20calling%20workflows/Table_S22_BioHansel_coverage_comparison_snippy_bam_mean_depths_Typhi_1000_datasets.txt)
- [Table S23 Genotyping results for BioHansel vs Snippy Accuracy Benchmarking Typhi.txt](./Results%20and%20Discussion/BioHansel's%20genotyping%20results%20have%20high%20concordance%20with%20traditional%20SNP%20calling%20workflows/Table_S23_Genotyping_results_for_BioHansel_vs_Snippy_Accuracy_Benchmarking_Typhi.txt)
- [Table S24 Genotyping results summary for BioHansel vs Snippy Accuracy Benchmarking Typhi.txt](./Results%20and%20Discussion/BioHansel's%20genotyping%20results%20have%20high%20concordance%20with%20traditional%20SNP%20calling%20workflows/Table_S24_Genotyping_results_summary_for_BioHansel_vs_Snippy_Accuracy_Benchmarking_Typhi.txt)
- [Table S25 Metadata for 1910 Typhi strains.txt](./Results%20and%20Discussion/BioHansel's%20genotyping%20results%20have%20high%20concordance%20with%20traditional%20SNP%20calling%20workflows/Table_S25_Metadata_for_1910_Typhi_strains.txt)
- [Table S26 BioHansel Results for 1910 Typhi strains.txt](./Results%20and%20Discussion/BioHansel's%20genotyping%20results%20have%20high%20concordance%20with%20traditional%20SNP%20calling%20workflows/Table_S26_BioHansel_Results_for_1910_Typhi_strains.txt)
- [Appendix IV Analysis of the 16 Typhi samples that failed BioHansel QC.pdf](./Results%20and%20Discussion/BioHansel's%20genotyping%20results%20have%20high%20concordance%20with%20traditional%20SNP%20calling%20workflows/Appendix%20IV%20Analysis_of_the_16_Typhi_samples_that_failed_BioHansel_QC.pdf)

### BioHansel detects contamination in synthetic WGS reads

- [Figure S3 Comparison of base detection results in synthetically contaminated datasets between BioHansel and Snippy.pdf](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Figure%20S3%20Comparison_of_base_detection_results_in_synthetically_contaminated_datasets_between_BioHansel_and_Snippy.pdf)
- [Table S27 Comparison base detection in mixed genomes for BioHansel and Snippy.txt](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Table_S27_Comparison_base_detection_in_mixed_genomes_for_BioHansel_and_Snippy.txt)
- [Table S28 Comparison of base detection at each target SNP position in mixed genomes for BioHansel and Snippy.xlsx](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Table_S28_Comparison_of_base_detection_at_each_target_SNP_position_in_mixed_genomes_for_BioHansel_and_Snippy.xlsx)
- [Table S29 Results Summary per scheme for comparison base detection in mixed genomes for BioHansel and Snippy.xlsx](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Table_S29_Results_Summary_per_scheme_for_comparison_base_detection_in_mixed_genomes_for_BioHansel_and_Snippy.xlsx)
- [Table S30 Results Summary comparison base detection in mixed genomes BioHansel and Snippy.pdf](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Table_S30_Results_Summary_comparison_base_detection_in_mixed_genomes_BioHansel_and_Snippy.pdf)
- [Table S31 BioHansel QA&QC results with synthetic contaminated SE datasets.xlsx](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Table_S31_BioHansel_QA%26QC_results_with_synthetic_contaminated_SE_datasets.xlsx)
- [Table S32 BioHansel QA&QC results with synthetic contaminated SH datasets.xlsx](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Table_S32_BioHansel_QA%26QC_results_with_synthetic_contaminated_SH_datasets.xlsx)
- [Table S33 BioHansel QA&QC results with synthetic contaminated Typhi datasets.xlsx](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Table_S33_BioHansel_QA%26QC_results_with_synthetic_contaminated_Typhi_datasets.xlsx)
- [Table S34 BioHansel QA&QC results with synthetic contaminated ST datasets.xlsx](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Table_S34_BioHansel_QA%26QC_results_with_synthetic_contaminated_ST_datasets.xlsx)
- [Table S35 BioHansel QA&QC results with synthetic contaminated MTB datasets.xlsx](./Results%20and%20Discussion/Quality%20assessment%20and%20quality%20control/Table_S35_BioHansel_QA%26QC_results_with_synthetic_contaminated_MTB_datasets.xlsx)

### BioHansel’s speed and memory performance compare favourably to traditional SNP-calling pipelines

- [Figure S4 BioHansel runtime on assemblies.pdf](./Results%20and%20Discussion/BioHansel's%20speed%20and%20memory%20performance/Figure%20S4%20BioHansel%20runtime%20on%20assemblies.pdf)
- [Figure S5 BioHansel memory usage on assemblies.pdf](./Results%20and%20Discussion/BioHansel's%20speed%20and%20memory%20performance/Figure%20S5%20BioHansel%20memory%20usage%20on%20assemblies.pdf)
- [Figure S6 BioHansel Runtime per assembly per core for different schemas.pdf](./Results%20and%20Discussion/BioHansel's%20speed%20and%20memory%20performance/Figure%20S6%20BioHansel%20Runtime%20per%20assembly%20per%20core%20for%20different%20schemas.pdf)
- [Table S36 Runtime on HPC computer cluster for BioHansel and Snippy vs WGS coverage.pdf](./Results%20and%20Discussion/BioHansel's%20speed%20and%20memory%20performance/Table%20S36%20Runtime%20on%20HPC%20computer%20cluster%20for%20BioHansel%20and%20Snippy%20vs%20WGS%20coverage.pdf)

