# BioHansel Manuscript Supplementary Data Repository.

This repository contains supplementary data for the BioHansel manuscript organized by section in which the supplementary data is referenced.

The directory structure of this repo follows the structure of the BioHansel manuscript.

## 8. Methods

### 8.1 SNP-based genotyping schemes

BioHansel scheme development documentation: https://bio-hansel.readthedocs.io/en/readthedocs/user-docs/genotyping_schemes.html#creating-a-genotyping-scheme

- [Additional_info_BioHansel_scheme_development_and_adaptation.docx](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/Additional_info_BioHansel_scheme_development_and_adaptation.docx)
- [Backbone_positions_5_schemes.xlsx](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/Backbone_positions_5_schemes.xlsx)

#### BioHansel Schemes

- [SE_scheme_v1.0.7.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/SE_scheme_v1.0.7.fasta)
- [SH_scheme_v0.5.0.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/SH_scheme_v0.5.0.fasta)
- [ST_scheme_v0.5.5.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/ST_scheme_v0.5.5.fasta)
- [tb_speciation_scheme_v1.0.5.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/tb_speciation_scheme_v1.0.5.fasta)
- [Typhi_scheme_v1.2.0.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/Typhi_scheme_v1.2.0.fasta)

#### Population_structure_backbones

- [Fig_Ia_Enteritidis_backbone_v1.0.7.pdf](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/Population_structure_backbones/Fig_Ia_Enteritidis_backbone_v1.0.7.pdf)
- [Fig_Ib_Heidelberg_backbone_v0.5.0.pdf](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/Population_structure_backbones/Fig_Ib_Heidelberg_backbone_v0.5.0.pdf)
- [Fib_Ic_Typhimurium_backbone_v0.5.5.pdf](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/Population_structure_backbones/Fib_Ic_Typhimurium_backbone_v0.5.5.pdf)
- [Fig_Id_Typhi_backbone_v1.2.0.pdf](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/Population_structure_backbones/Fig_Id_Typhi_backbone_v1.2.0.pdf)
- [Fig_Ie_MTB_backbone_v1.0.5.pdf](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/Population_structure_backbones/Fig_Ie_MTB_backbone_v1.0.5.pdf)
- [Caption_trees_Fig_Ia-e.txt](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/Population_structure_backbones/Caption_trees_Fig_Ia-e.txt)

### 8.2 BioHansel

### Examples of BioHansel output files

- [BioHansel_Simple_Summary_genome_NC_011083_tech_results.tab](./8.%20Methods/8.2%20BioHansel/8.2%20Examples%20of%20BioHansel%20output%20files/BioHansel_Simple_Summary_genome_NC_011083_tech_results.tab)
- [BioHansel_Results_Summary_genome_NC_011083_results.tab](./8.%20Methods/8.2%20BioHansel/8.2%20Examples%20of%20BioHansel%20output%20files/BioHansel_Results_Summary_genome_NC_011083_results.tab)
- [BioHansel_Detailed_K-mer_Results_genome_NC_011083_match_results.tab](./8.%20Methods/8.2%20BioHansel/8.2%20Examples%20of%20BioHansel%20output%20files/BioHansel_Detailed_K-mer_Results_genome_NC_011083_match_results.tab)
- [BioHansel_Simple_Summary_unassembled_reads_SH_outbreaks_tech_results.tab](./8.%20Methods/8.2%20BioHansel/8.2%20Examples%20of%20BioHansel%20output%20files/BioHansel_Simple_Summary_unassembled_reads_SH_outbreaks_tech_results.tab)
- [BioHansel_Results_Summary_unassembled_reads_SH_outbreaks_results.tab](./8.%20Methods/8.2%20BioHansel/8.2%20Examples%20of%20BioHansel%20output%20files/BioHansel_Results_Summary_unassembled_reads_SH_outbreaks_results.tab)
- [BioHansel_Detailed_K-mer_Results_unassembled_reads_SH_outbreaks_match_results.tab](./8.%20Methods/8.2%20BioHansel/8.2%20Examples%20of%20BioHansel%20output%20files/BioHansel_Detailed_K-mer_Results_unassembled_reads_SH_outbreaks_match_results.tab)

#### 8.2.3 BioHansel metadata tables

- [tb_speciation_scheme_v1.0.5_metadata_Coll_et_al_genotypes.txt](./8.%20Methods/8.2%20BioHansel/8.2.3%20BioHansel%20metadata%20tables/tb_speciation_scheme_v1.0.5_metadata_Coll_et_al_genotypes.txt)
- [Typhi_scheme_v1.2.0_metadata_Wong_et_al_genotypes.txt](./8.%20Methods/8.2%20BioHansel/8.2.3%20BioHansel%20metadata%20tables/Typhi_scheme_v1.2.0_metadata_Wong_et_al_genotypes.txt)

#### 8.2.4 Source Code

Source code available at https://github.com/phac-nml/biohansel/

### 8.3 Analysis of retrospective outbreaks with BioHansel

- [Table_S1_SH_Outbreaks_Bekal_and_coll.tab](./8.%20Methods/8.3%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/Table_S1_SH_Outbreaks_Bekal_and_coll.tab)
- [SH_ref_SL476_NC_011083.1.fasta](./8.%20Methods/8.3%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SH_ref_SL476_NC_011083.1.fasta)

### 8.4 Validating BioHansel genotyping results with traditional SNP calling workflows

#### 8.4.1.1 Real Illumina Dataset of *S*. Typhi

- [Table_S2_List_of_1000_Salmonella_Typhi_Accessions.txt](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.1%20Real%20Illumina%20Dataset%20of%20S.%20Typhi/Table_S2_List_of_1000_Salmonella_Typhi_Accessions.txt)

#### 8.4.1.2 Validating BioHansel using real *S*. Typhi WGS data

See https://github.com/peterk87/nf-biohansel-snippy-comparison/blob/master/main.nf

#### 8.4.1.3 Generation of artificially contaminated synthetic datasets for 5 pathogens

- [Table_S3_List_of_reference_genomes.txt](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.3%20Generation%20of%20artificially%20contaminated%20synthetic%20datasets%20for%205%20pathogens/Table_S3_List_of_reference_genomes.txt)
- [Worksheet 1 Bash and Seqtk Commands.xlsx](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.3%20Generation%20of%20artificially%20contaminated%20synthetic%20datasets%20for%205%20pathogens/Worksheet%201%20Bash%20and%20Seqtk%20Commands.xlsx)

#### 8.4.1.4 SNP detection on artificially contaminated synthetic datasets

- [AE006468.2_ref_ST_LT2.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/AE006468.2_ref_ST_LT2.fasta)
- [CP012921.1_ref_SH.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/CP012921.1_ref_SH.fasta)
- [NC_000962.3_ref_MTB_H37Rv.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/NC_000962.3_ref_MTB_H37Rv.fasta)
- [NC_003198.1_ref_Typhi_CT18.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/NC_003198.1_ref_Typhi_CT18.fasta)
- [NC_011294.1_ref_SE_P125109.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/NC_011294.1_ref_SE_P125109.fasta)

### 8.5 Benchmarking the BioHansel QA/QC module

#### 8.5.1 Selecting representative datasets

- [Table_S4_List_of_MTB_accessions_for_generation_of_contaminated_datasets.txt](./8.%20Methods/8.5%20Benchmarking%20the%20BioHansel%20QA&QC/8.5.1%20Selecting%20representative%20datasets/Table_S4_List_of_MTB_accessions_for_generation_of_contaminated_datasets.txt)
- [Table_S5_List_of_SE_accessions_for_generation_of_contaminated_datasets.txt](./8.%20Methods/8.5%20Benchmarking%20the%20BioHansel%20QA&QC/8.5.1%20Selecting%20representative%20datasets/Table_S5_List_of_SE_accessions_for_generation_of_contaminated_datasets.txt)
- [Table_S6_List_of_ST_accessions_for_generation_of_contaminated_datasets.txt](./8.%20Methods/8.5%20Benchmarking%20the%20BioHansel%20QA&QC/8.5.1%20Selecting%20representative%20datasets/Table_S6_List_of_ST_accessions_for_generation_of_contaminated_datasets.txt)
- [Table_S7_List_of_SH_accessions_for_generation_of_contaminated_datasets.txt](./8.%20Methods/8.5%20Benchmarking%20the%20BioHansel%20QA&QC/8.5.1%20Selecting%20representative%20datasets/Table_S7_List_of_SH_accessions_for_generation_of_contaminated_datasets.txt)
- [Table_S8_List_of_Typhi_accessions_for_generation_of_contaminated_datasets.txt](./8.%20Methods/8.5%20Benchmarking%20the%20BioHansel%20QA&QC/8.5.1%20Selecting%20representative%20datasets/Table_S8_List_of_Typhi_accessions_for_generation_of_contaminated_datasets.txt)

#### 8.5.2 Generation of artificially contaminated datasets

- [contaminate.sh](./8.%20Methods/8.5%20Benchmarking%20the%20BioHansel%20QA&QC/8.5.2%20Generation%20of%20artificially%20contaminated%20datasets/contaminate.sh)
- [create_reads.sh](./8.%20Methods/8.5%20Benchmarking%20the%20BioHansel%20QA&QC/8.5.2%20Generation%20of%20artificially%20contaminated%20datasets/create_reads.sh)
- [Additional_info_Generation_of_artificially_contaminated_datasets.pdf](./8.%20Methods/8.5%20Benchmarking%20the%20BioHansel%20QA&QC/8.5.2%20Generation%20of%20artificially%20contaminated%20datasets/Additional_info_Generation_of_artificially_contaminated_datasets.pdf)

### 8.6 Benchmarking BioHansel performance for computational speed and memory usage

See https://github.com/peterk87/nf-biohansel-sra-benchmark for BioHansel and Snippy benchmarking workflow

#### 8.6.1 Datasets

- [Table_S9_List_of_1017_Salmonella_Enteritidis_strains_NextFlow_workflow.txt](./8.%20Methods/8.6%20Benchmarking%20BioHansel%20performance%20for%20computational%20speed%20and%20memory%20usage/8.6.1%20Datasets/Table_S9_List_of_1017_Salmonella_Enteritidis_strains_NextFlow_workflow.txt)
- [Table_S10_List_of_1000_Salmonella_Enteritidis_strains.tab](./8.%20Methods/8.6%20Benchmarking%20BioHansel%20performance%20for%20computational%20speed%20and%20memory%20usage/8.6.1%20Datasets/Table_S10_List_of_1000_Salmonella_Enteritidis_strains.tab)
- [Table_S11_List_of_1000_Salmonella_Heidelberg_strains.tab](./8.%20Methods/8.6%20Benchmarking%20BioHansel%20performance%20for%20computational%20speed%20and%20memory%20usage/8.6.1%20Datasets/Table_S11_List_of_1000_Salmonella_Heidelberg_strains.tab)
- [Table_S12_List_of_1000_Salmonella_Typhi_strains.tab](./8.%20Methods/8.6%20Benchmarking%20BioHansel%20performance%20for%20computational%20speed%20and%20memory%20usage/8.6.1%20Datasets/Table_S12_List_of_1000_Salmonella_Typhi_strains.tab)
- [Table_S13_List_of_1000_Salmonella_Typhimurium_strains.tab](./8.%20Methods/8.6%20Benchmarking%20BioHansel%20performance%20for%20computational%20speed%20and%20memory%20usage/8.6.1%20Datasets/Table_S13_List_of_1000_Salmonella_Typhimurium_strains.tab)

#### 8.6.2 Generation of artificial sequencing data representing various genome coverage

- [AE006468.2_ref_ST_LT2.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/AE006468.2_ref_ST_LT2.fasta)
- [CP012921.1_ref_SH.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/CP012921.1_ref_SH.fasta)
- [NC_000962.3_ref_MTB_H37Rv.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/NC_000962.3_ref_MTB_H37Rv.fasta)
- [NC_003198.1_ref_Typhi_CT18.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/NC_003198.1_ref_Typhi_CT18.fasta)
- [NC_011294.1_ref_SE_P125109.fasta](./8.%20Methods/8.4%20Validating%20BioHansel%20genotyping%20results%20with%20traditional%20SNP%20calling%20workflows/8.4.1.4%20SNP%20detection%20on%20artificially%20contaminated%20synthetic%20datasets/NC_011294.1_ref_SE_P125109.fasta)

#### 8.6.3 Determination of runtime and memory usage

- [py-biohansel-vs-snippy-benchmarking-results.html](./8.%20Methods/8.6%20Benchmarking%20BioHansel%20performance%20for%20computational%20speed%20and%20memory%20usage/8.6.3%20Determination%20of%20runtime%20and%20memory%20usage/py-biohansel-vs-snippy-benchmarking-results.html)
- [py-biohansel-vs-snippy-benchmarking-results.ipynb](./8.%20Methods/8.6%20Benchmarking%20BioHansel%20performance%20for%20computational%20speed%20and%20memory%20usage/8.6.3%20Determination%20of%20runtime%20and%20memory%20usage/py-biohansel-vs-snippy-benchmarking-results.ipynb)
- [trace.csv](./8.%20Methods/8.6%20Benchmarking%20BioHansel%20performance%20for%20computational%20speed%20and%20memory%20usage/8.6.3%20Determination%20of%20runtime%20and%20memory%20usage/trace.csv)

## 9. Results and Discussion

### 9.1 Analysis of retrospective outbreaks with BioHansel

- [Table_S1_SH_Outbreaks_Bekal_and_coll.tab](./8.%20Methods/8.6%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/Table_S12_SH_Outbreaks_Bekal_and_coll.tab)
- [Table S14 S Heidelberg Backbone positions BioHansel scheme v0.5.0.xlsx](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/Table%20S33%20S%20Heidelberg%20Backbone%20positions%20BioHansel%20scheme%20v0.5.0.xlsx)
- [Table S15 SH outbreaks SNV distance matrix.xlsx](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/Table%20S34%20SH%20outbreaks%20SNV%20distance%20matrix.xlsx)


#### BioHansel_results_SH_outbreaks

- [BioHansel_raw_datasets_match_results_SH_outbreaks.xlsx](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/BioHansel_results_SH_outbreaks/BioHansel_raw_datasets_match_results_SH_outbreaks.xlsx)
- [BioHansel_raw_datasets_results_Sadjia_SH_outbreaks.xlsx](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/BioHansel_results_SH_outbreaks/BioHansel_raw_datasets_results_Sadjia_SH_outbreaks.xlsx)
- [BioHansel_raw_datasets_tech_results_SH_outbreaks.xlsx](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/BioHansel_results_SH_outbreaks/BioHansel_raw_datasets_tech_results_SH_outbreaks.xlsx)

#### SNVPhyl_run_SH_outbreaks

![Figure_SH_outbreaks_v4.jpg](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Figure_SH_outbreaks_v4.jpg)

- [Figure_SH_outbreaks_v4.pptx](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Figure_SH_outbreaks_v4.pptx)
- [Galaxy545-[combined_invalid_positions].data](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Galaxy545-[combined_invalid_positions].data)
- [Galaxy546-[SH_outbreak_analysis-snvTable.tsv].tabular](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Galaxy546-[SH_outbreak_analysis-snvTable.tsv].tabular)
- [Galaxy547-[SH_outbreak_analysis-snvalign.fasta].fasta](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Galaxy547-[SH_outbreak_analysis-snvalign.fasta].fasta)
- [Galaxy549-[SH_outbreak_analysis-vcf2core.tsv].tabular](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Galaxy549-[SH_outbreak_analysis-vcf2core.tsv].tabular)
- [Galaxy550-[SH_outbreak_analysis-snvAlignment.phy].phylip](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Galaxy550-[SH_outbreak_analysis-snvAlignment.phy].phylip)
- [Galaxy551-[SH_outbreak_analysis-filterStats.txt].tabular](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Galaxy551-[SH_outbreak_analysis-filterStats.txt].tabular)
- [Galaxy552-[SH_outbreak_analysis-phylogeneticTree.newick].nhx](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Galaxy552-[SH_outbreak_analysis-phylogeneticTree.newick].nhx)
- [Galaxy553-[SH_outbreak_analysis-phylogeneticTreeStats.txt].txt](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Galaxy553-[SH_outbreak_analysis-phylogeneticTreeStats.txt].txt)
- [Galaxy554-[SH_outbreak_analysis-snvMatrix.tsv].tabular](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/Galaxy554-[SH_outbreak_analysis-snvMatrix.tsv].tabular)
- [SH_outbreak_analysis-results.zip](./9.%20Results%20and%20Discussion/9.4%20Analysis%20of%20retrospective%20outbreaks%20with%20BioHansel/SNVPhyl_run_SH_outbreaks/SH_outbreak_analysis-results.zip)


### 9.2 BioHansel concordance with traditional SNP calling workflows

- [Analysis_of_the_16_Typhi_samples_that_failed_BioHansel_QC.docx](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/Analysis_of_the_16_Typhi_samples_that_failed_BioHansel_QC.docx)
- [Table S16 BioHansel_Coverage_Comparison_1000_Typhi_datasets.xlsx](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/Table%20S13%20BioHansel_Coverage_Comparison_1000_Typhi_datasets.xlsx)
- [Tables S17 and S18 BioHansel_vs_Snippy_Accuracy_Benchmarking_Typhi.xlsx](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/Tables%20S14%20and%20S15%20BioHansel_vs_Snippy_Accuracy_Benchmarking_Typhi.xlsx)
- [Tables S19 to S21 Comparison base detection in mixed genomes for BioHansel and Snippy.xlsx](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/Tables%20S16%20to%20S18%20Comparison%20base%20detection%20in%20mixed%20genomes%20for%20BioHansel%20and%20Snippy.xlsx)
- [Table S22_and_Figure_S1.docx](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/Table%20S19_and_Figure_S1.docx)
- [Tables S23 and S24 Metadata 1910 Typhi strains and BioHansel Results.xlsx](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/Tables%20S20%20and%20S21%20Metadata%201910%20Typhi%20strains%20and%20BioHansel%20Results.xlsx)


#### K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC

- [all_in_one_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/all_in_one_kmer_results.tsv)
- [ERR204256_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR204256_kmer_results.tsv)
- [ERR213231_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR213231_kmer_results.tsv)
- [ERR213233_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR213233_kmer_results.tsv)
- [ERR213234_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR213234_kmer_results.tsv)
- [ERR213249_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR213249_kmer_results.tsv)
- [ERR213251_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR213251_kmer_results.tsv)
- [ERR213262_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR213262_kmer_results.tsv)
- [ERR279153_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR279153_kmer_results.tsv)
- [ERR279164_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR279164_kmer_results.tsv)
- [ERR279165_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR279165_kmer_results.tsv)
- [ERR279178_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR279178_kmer_results.tsv)
- [ERR279191_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR279191_kmer_results.tsv)
- [ERR326600_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR326600_kmer_results.tsv)
- [ERR340783_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR340783_kmer_results.tsv)
- [ERR357603_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR357603_kmer_results.tsv)
- [ERR360616_kmer_results.tsv](./9.%20Results%20and%20Discussion/9.1%20BioHansel%20concordance%20with%20traditional%20SNP%20calling%20workflows/K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC/ERR360616_kmer_results.tsv)

### 9.3 QA-QC

#### 9.3.1 BioHansel QC module detects contamination in synthetic WGS reads

- [Tables S25-S29 Contaminated_datasets_BioHansel_results_with_charts.xlsx](./9.%20Results%20and%20Discussion/9.2%20QA-QC/9.2.1%20Detection%20of%20contamination%20in%20synthetic%20raw%20reads%20by%20the%20BioHansel%20QC%20module/Tables%20S22-S26%20Contaminated_datasets_BioHansel_results_with_charts.xlsx)

### 9.4 Performance: comparing BioHansel runtime and memory usage with a traditional SNP genotyping workflow


- [Table S31_and_Figures_S2_to_S4.docx](./9.%20Results%20and%20Discussion/9.3%20Performance_%20comparing%20BioHansel%20runtime%20and%20memory%20usage%20with%20a%20traditional%20SNP%20genotyping%20workflow/Table%20S31_and_Figures_S2_to_S4.docx)
- [Tables S30, S31, and S34 Run_Time_Memory_Stats_HPC_cluster_BioHansel_vs_Snippy.xlsx](./9.%20Results%20and%20Discussion/9.3%20Performance_%20comparing%20BioHansel%20runtime%20and%20memory%20usage%20with%20a%20traditional%20SNP%20genotyping%20workflow/Tables%20S27,%20S28,%20and%20S31%20Run_Time_Memory_Stats_HPC_cluster_BioHansel_vs_Snippy.xlsx)
- [Tables S32 and S33 Local_Biohansel_runtime_results.xlsx](./9.%20Results%20and%20Discussion/9.3%20Performance_%20comparing%20BioHansel%20runtime%20and%20memory%20usage%20with%20a%20traditional%20SNP%20genotyping%20workflow/Tables%20S29%20and%20S30%20Local_Biohansel_runtime_results.xlsx)

