# BioHansel Manuscript Supplementary Data Repository.

This repository contains supplementary data for the BioHansel manuscript organized by section in which the supplementary data is referenced.

The directory structure of this repo follows the structure of the BioHansel manuscript.

## 8. Methods

### 8.1 SNP-based genotyping schemes

#### BioHansel Schemes

- [SE_scheme_v1.0.7.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/SE_scheme_v1.0.7.fasta)
- [SH_scheme_v0.5.0.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/SH_scheme_v0.5.0.fasta)
- [ST_scheme_v0.5.5.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/ST_scheme_v0.5.5.fasta)
- [tb_speciation_scheme_v1.0.5.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/tb_speciation_scheme_v1.0.5.fasta)
- [Typhi_scheme_v1.2.0.fasta](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/BioHansel%20Schemes/Typhi_scheme_v1.2.0.fasta)

#### Population_structure_backbones

- [Backbone_positions_5_schemes.xlsx](./8.%20Methods/8.1%20SNP-based%20genotyping%20schemes/Backbone_positions_5_schemes.xlsx)

### 8.2 BioHansel

#### 8.2.3 BioHansel metadata tables

#### 8.2.4 Source Code

Source code available at https://github.com/phac-nml/biohansel/

### 8.2 Examples of BioHansel output files

### 8.3 Benchmarking BioHansel with traditional SNP calling workflows

#### 8.3.1.1 Real Illumina Dataset of S. Typhi

#### 8.3.1.2 Comparing Snippy and BioHansel for SNP Calling in real Illumina datasets of S. Typhi

#### 8.3.1.3 Generation of artificially contaminated synthetic datasets for 5 pathogens

#### 8.3.1.4 SNP detection on artificially contaminated synthetic datasets

### 8.4 Benchmarking the BioHansel QA-QC module

#### 8.4.1 Selecting representative datasets

#### 8.4.2 Generation of artificially contaminated datasets

### 8.5 Benchmarking BioHansel performance for computational speed and memory usage

#### 8.5.1 Datasets

#### 8.5.2 Generation of artificial sequencing data representing various genome coverage

#### 8.5.3 Determination of runtime and memory usage

### 8.6 Analysis of retrospective outbreaks with BioHansel

## 9. Results and Discussion

### 9.1 BioHansel concordance with traditional SNP calling workflows

#### K-mer_match_results_for_16_Typhi_samples_that_failed_BioHansel_QC

### 9.2 QA-QC

#### 9.2.1 Detection of contamination in synthetic raw reads by the BioHansel QC module

### 9.3 Performance_ comparing BioHansel runtime and memory usage with a traditional SNP genotyping workflow

### 9.4 Analysis of retrospective outbreaks with BioHansel

#### BioHansel_results_SH_outbreaks

#### SNVPhyl_run_SH_outbreaks

